src/app/pages/data-protection/replication/replication-list/replication-list.component.ts

```diff
@@ -1,306 +1,280 @@
-      } else {
-        this.matDialog.open(ShowLogsDialogComponent, { data: row.job });
-      }
     } else {
-      this.dialogService.warn(helptextGlobal.noLogDialog.title, helptextGlobal.noLogDialog.message);
+      this.chainedSlideInService.pushComponent(ReplicationWizardComponent, true)
+        .pipe(
+          filter((response) => !!response.response),
+          untilDestroyed(this),
+        ).subscribe(() => this.getReplicationTasks());
     }
   }
 
-  doEdit(id: number): void {
-    const replication = this.entityList.rows.find((row) => row.id === id);
-    const closer$ = this.chainedSlideInService.pushComponent(ReplicationFormComponent, true, replication);
-    closer$.pipe(
-      filter((response) => !!response.response),
+  doDelete(row: ReplicationTask): void {
+    this.dialogService.confirm({
+      title: this.translate.instant('Confirmation'),
+      message: this.translate.instant('Delete Replication Task <b>"{name}"</b>?', {
+        name: row.name,
+      }),
+    }).pipe(
+      filter(Boolean),
+      switchMap(() => this.ws.call('replication.delete', [row.id]).pipe(this.appLoader.withLoader())),
       untilDestroyed(this),
-    ).subscribe(() => this.entityList.getData());
-  }
-
-  onCheckboxChange(row: ReplicationTaskUi): void {
-    this.ws.call('replication.update', [row.id, { enabled: row.enabled }]).pipe(untilDestroyed(this)).subscribe({
-      next: (task) => {
-        row.enabled = task.enabled;
-        if (!task) {
-          row.enabled = !row.enabled;
-        }
+    ).subscribe({
+      next: () => {
+        this.getReplicationTasks();
       },
-      error: (err: unknown) => {
-        row.enabled = !row.enabled;
+      error: (err) => {
         this.dialogService.error(this.errorHandler.parseError(err));
       },
     });
   }
 
-  doAdd(): void {
-    const closer$ = this.chainedSlideInService.pushComponent(ReplicationWizardComponent, true);
-    closer$.pipe(
-      filter((response) => !!response.response),
-      untilDestroyed(this),
-    ).subscribe(() => this.entityList.getData());
+  onListFiltered(query: string): void {
+    this.filterString = query.toLowerCase();
+    this.dataProvider.setRows(this.replicationTasks.filter((task) => {
+      return task.name.includes(this.filterString);
+    }));
+  }
+
+  columnsChange(columns: typeof this.columns): void {
+    this.columns = [...columns];
+    this.cdr.detectChanges();
+    this.cdr.markForCheck();
+  }
+
+  downloadKeys(row: ReplicationTask): void {
+    const fileName = `${row.name}_encryption_keys.json`;
+    this.ws.call('core.download', ['pool.dataset.export_keys_for_replication', [row.id], fileName])
+      .pipe(this.appLoader.withLoader(), untilDestroyed(this))
+      .subscribe({
+        next: ([, url]) => {
+          const mimetype = 'application/json';
+          this.storage.streamDownloadFile(url, fileName, mimetype).pipe(untilDestroyed(this)).subscribe({
```

Comment:
Prefer using `switchMaps` pipes to keep the flow more readable. Something like the following
`this.ws.call(...).pipe( ... , switchMap(([, url]) => { ... ; return this.storage.streamDownloadFile(...); ... }))`